engine/src/main/java/io/camunda/zeebe/streamprocessor/StreamProcessor.java

```diff
@@ -147,6 +150,11 @@ public static StreamProcessorBuilder builder() {
     return new StreamProcessorBuilder();
   }
 
+  @Deprecated // only used for tests
```

Comment:
:upside_down_face: We're starting to see this more and more, might be time to introduce our own annotation, much like Google's `@VisibleForTesting` or the likes. This is a very soft suggestion :)

Reply:
I'll keep it in the back of my head. Still hoping that the methods can soon be removed during refactoring, but I agree it's on the threshold to think about it.